File: 2236-root-equals-sum-of-children/2236-root-equals-sum-of-children.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def checkTree(self, root):
        if root is None:
            return True
        sum = 0
        if root.left is not None:
            sum+=root.left.val
        if root.right is not None:
            sum+=root.right.val
            
        return sum==root.val
            
        
        

File: 2236-root-equals-sum-of-children/2236-root-equals-sum-of-children.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun checkTree(root: TreeNode?): Boolean {
        
//         basecase
        if (root == null)
            return true
        
        if(root.left!= null && root.right != null)
        {
            return (root.`val` == root.left.`val` + root.right.`val`)
        }
        return false
        
    }
}